### PR TITLE
[DAG] Adding randomness field in dag node

### PR DESCRIPTION
**Pull request by danielxiangzl**
### Description
Reserve the field for randomness shares in the dag node, to make future randomness upgrades compatible. 
 
### Test Plan

Existing tests.

### CODE REVIEW

File: consensus/src/dag/types.rs [hunk 1]
```diff
@@ -46,6 +46,7 @@ struct NodeWithoutDigest<'a> {
     timestamp: u64,
     payload: &'a Payload,
     parents: &'a Vec<NodeCertificate>,
+    shares: &'a Vec<u8>, // serialized bytes of randomness shares
```
Comment (ibalajiarun) [inline consensus/src/dag/types.rs:49]: This should be an `Option<>`?

Comment (ibalajiarun) [reply consensus/src/dag/types.rs:49]: Also, annotate with `#[serde(with = "serde_bytes")]` (lessons from last week 😬 )

**Review by zekun000 — APPROVED** [review]

File: consensus/src/dag/types.rs [hunk 2]
```diff
@@ -38,6 +38,18 @@ impl TDAGMessage for CertifiedAck {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, CryptoHasher, Debug, PartialEq)]
+pub enum Extensions {
+    Empty,
```
Comment (ibalajiarun) [inline consensus/src/dag/types.rs:43]: nit: None is probably more appropriate.

```suggestion
    None,
```

**Review by ibalajiarun — APPROVED** [review]
So, what's the idea with the enums? Will there be versioned structs as we add more fields later? Like V1(...), V2(...), etc?

**Comment by zekun000** [comment]
@ibalajiarun Yes, it's basically a versioning for more fields.

**Pull request merged**